larq_compute_engine/mlir/transforms/prepare_patterns_common.td

```diff
@@ -4,9 +4,56 @@ include "tensorflow/compiler/mlir/tensorflow/ir/tf_ops.td"
 include "larq_compute_engine/mlir/ir/lce_ops.td"
 include "larq_compute_engine/mlir/transforms/op_removal_patterns.td"
 
+class ConstantValue<string val> : AttrConstraint<CPred<"IsConstantValue($_self, " # val # ")">>;
 
-// This relies on implementation details of larq.math.sign. We should make
-// this more general in the future
+def CreateTFBroadcastToOp : NativeCodeCall<
+    "$_builder.create<TF::BroadcastToOp>("
+        "$0.getLoc(),"
+        "RankedTensorType::get("
+            "$0.getType().cast<RankedTensorType>().getShape(),"
+            "getElementTypeOrSelf($1.getType())),"
+        "$1,"
+        "$2)">;
```

Comment:
@lgeiger after running into the hell of trying to insert nested ops with tablegen, I eventually resorted to this rather contrived native code call. This took ages to get right, but it does now work correctly with broadcasting any of the three select op arguments (verified with new MLIR test cases) :tada:

Reply:
What does this broadcasting allow us to match against that we would not get without it?

Reply:
Basically the `tf.where` op supports broadcasting all three of it's inputs (the condition, lhs, and rhs). So this would allow, for example, `tf.where(<tensor shape=(5, 32)>, <ones shape=(5, 5, 32)>, <-ones shape=(5, 5, 32)>)`. That of course would not really make much sense from a ML perspective, but since `tf.where` supports such constructions I don't think it hurts to support it.